addons/purchase_stock/tests/test_replenish_wizard.py

```diff
@@ -585,13 +555,129 @@ def test_purchase_order_uom(self):
+
+        self.assertEqual(po.partner_id, self.vendor)
+        self.assertEqual(po.order_line.price_unit, 0.0)  # Should not take price from pricelist
+        self.assertEqual(po.currency_id, self.env.company.currency_id)
+        self.assertEqual(po.id, po_1.id)  # Should not create a new PO
+        self.assertEqual(po.order_line.product_qty, 2.0)  # Should add to previous PO
+
+        # Test replenishing from partner on pricelist
+        replenish_wizard.partner_id = vendor2.id
+        generated_picking = replenish_wizard.launch_replenishment()
+        po = self._get_po_id_from_picking(generated_picking)
+
+        self.assertEqual(po.partner_id, vendor2)  # Should take pricelist
+        self.assertEqual(po.order_line.price_unit, 140.0)
+
+    def test_buy_replenish_name_search(self):
+        """ On replenishement with buy route, partner name_search suppliers should display supplier first and then contacts"""
+        # With highlight flag + product_id -> seller should be at the top
+        res_highlight = (
+            self.env['res.partner']
+            .with_context(highlight_supplier=1, product_id=self.product1.id)
+            .name_search('', limit=10)
+        )
+        self.assertEqual(res_highlight[0][0], self.vendor.id)
+
+        # Check edge case of vendors starting with eg. Z are still displayed at top even not part of limit
+        res_highlight = (
+            self.env['res.partner']
+            .with_context(highlight_supplier=1, product_id=self.product1.id)
+            .name_search('', limit=2)  # Simulate a lot of contacts with limit 2
+        )
+        self.assertEqual(res_highlight[0][0], self.vendor.id)
+
+        # Finally check that without the flag we have the default behaviour
+        res_default = self.env['res.partner'].name_search('', limit=10)
+        self.assertNotEqual(res_default[0][0], self.vendor.id)  # Vendor should be first
+
+    def test_buy_replenish_defaults(self):
+        """ The partner and scheduled date fields of the replenish wizard
+            with the BUY route should select to the best pricelist for parameters
+            UOM and quantity on open form and on_change 'route_id' & 'quantity' """
+        product = self.env['product.product'].create({
+            'name': "Furniture Service",
+            'is_storable': True,
+            'route_ids': [Command.link(self.route_buy.id)],
+        })
+        self.env['product.supplierinfo'].create([{
+            'partner_id': self.partner.id,
+            'product_id': product.id,
+            'price': 140.0,
+            'delay': 10,
+        }, {
+            'partner_id': self.partner.id,
+            'product_id': product.id,
+            'price': 100.0,
+            'min_qty': 10,
+            'delay': 12,
+        }, {
+            'partner_id': self.partner_1.id,
+            'product_id': product.id,
+            'price': 50.0,
+            'min_qty': 100,
+        }])
+
+        with freeze_time("2023-01-01"):
```

Comment:
You can tag the test with `@freeze_time('2023-01-01')` (on top of the definition of the test method) so you can remove this `with` and gain an indent on all following lines :)
I think if you want to run a test at a precise date, using the tag is better, while `with freeze_time` is better when you want to try only a part of your test at the given date